### DISCUSSION TITLE
[proposal] model context protocol server for Pocket base AI <> Pocketbase

### DISCUSSION THREAD

**Discussion by jeffscottward** [pocketbase/pocketbase#6134] [Q&A]
I've briefly made a request to the project owner previously about some sort of AI integration, but it was neither his focus nor in the scope of what he has been doing and probably neither is this, but I wanted to put it on everybody's radar. 

Pocketbase should be able to talk to AI one way or another and anthropic released the model context protocol with API endpoint support for being able to do requests where direct SQL queries are not available such as GitHub Slack and other api providers. This module would probably be its own repo somewhere else, but I thought it was relevant to mention it here. 

It is just simply not feasible for pocketbase to be only a point-and-click web UI for its entire existence that's just not going to happen.

**Comment by jeffscottward** [comment]
https://github.com/DynamicEndpoints/pocketbase-mcp-server

Looks like it's maybe already a done deal?

**Reply by gevera** [reply]
It would be nice if it would be official

**Comment by louislili** [comment]
need mcp

**Comment by Sleepful** [comment]
Just print the schema to a text file and feed it to the LLM, e.g:

```js
import PocketBase from 'pocketbase'
const pb = new PocketBase('http://127.0.0.1:8090');

const userData = await pb.collection('_superusers').authWithPassword('x@email', 'x');


// Fetch all collections as an admin user
const collections = await pb.collections.getList();

console.log("// Data generated with fetchColls.js script\n\n")

// Log the details (example)
collections.items.forEach(collection => {
  if (collection.name.startsWith("_")) return
  console.log(`Collection: ${collection.name}`);
  console.log('Fields:'); // 'schema' contains the column definitions
  collection.fields.forEach(e => {
    console.log(`  - Name: ${e.name},  Type: ${e.type}`);
  })
});
```

adjust to your liking
